File: authlib/Authentication.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Security.Cryptography;
using System.Text;
using System.Threading.Tasks;

namespace authlib
{
    public class Authentication
    {
        public static bool Verify(string UserID, string Code)
        {
            if (Code == CurrentCode(UserID))
                return true;
            else
                return false;
        }

        public static string CurrentCode(string UserID)
        {
            long seconds = (int)(DateTime.UtcNow - new DateTime(1970, 1, 1)).TotalSeconds;
            string Key = UserID + (seconds - (seconds % 16));
            string hash = EHashing.Hash(Key);
            string Code = "";
            for (int i = 0; i < 8; i++)
            {
                Code += "" + long.Parse("" +
                    hash.ToCharArray()[8 * i] +
                    hash.ToCharArray()[8 * i + 1] +
                    hash.ToCharArray()[8 * i + 2] +
                    hash.ToCharArray()[8 * i + 3] +
                    hash.ToCharArray()[8 * i + 4] +
                    hash.ToCharArray()[8 * i + 5] +
                    hash.ToCharArray()[8 * i + 6] +
                    hash.ToCharArray()[8 * i + 7], System.Globalization.NumberStyles.HexNumber) % 10;
            }
            return Code;
        }
    }
}

File: authlib-java/src/authlib/Authentication.java
package authlib;

public class Authentication {
	public static boolean Verify(String UserID, String Code) {
		if (Code == CurrentCode(UserID))
			return true;
		else
			return false;
	}
	
	public static String CurrentCode(String UserID) {
		long seconds = java.time.Instant.now().getEpochSecond();
		String Key = UserID + (seconds - (seconds % 16));
		String hash = EHashing.Hash(Key);
		String Code = "";
        for (int i = 0; i < 8; i++)
        {
            Code += "" + Long.parseLong("" +
                hash.toCharArray()[8 * i] +
                hash.toCharArray()[8 * i + 1] +
                hash.toCharArray()[8 * i + 2] +
                hash.toCharArray()[8 * i + 3] +
                hash.toCharArray()[8 * i + 4] +
                hash.toCharArray()[8 * i + 5] +
                hash.toCharArray()[8 * i + 6] +
                hash.toCharArray()[8 * i + 7], 16) % 10;
        }
        return Code;
	}
}
